.github/utils/check_version.py

```diff
@@ -5,16 +5,38 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-import os
 import sys
-from importlib.util import find_spec
+import tomllib
+from pathlib import Path
 
-from setuptools.config.expand import StaticModule
+from hatchling.version.core import VersionFile
 
 
-package = 'indico'
-sys.path.insert(0, os.getcwd())
-version = StaticModule(package, find_spec(package)).__version__
+def _get_pyproject_version(base_dir: Path):
+    data = tomllib.loads((base_dir / 'pyproject.toml').read_text())
+    # try a static version
+    try:
+        return data['project']['version']
+    except KeyError:
+        pass
```

Comment:
This is not absolutely necessary, but I wanted to keep it similar yo the util we use in the i18n cli. Also like this I can just copy it into other packages if needed ;)